---
title: "GeneOnt"
output: html_document
date: "2024-12-27"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("DESeq2")
#install.packages("fastmap")

#BiocManager::install("clusterProfiler")
#BiocManager::install("org.Hs.eg.db")
#BiocManager::install("AnnotationDbi")

#install.packages("org.Hs.eg.db")
library(dplyr)
library(ggplot2)
library(DESeq2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(AnnotationDbi)
library(clusterProfiler)
library(pheatmap)
library(stringr)


gene_template <- read_excel("/Users/dylan/Documents/CRISPR Screen data/WT/WT_CRISPR_finalAnalysis.xlsx", sheet = "Raw_Data")

#USE MODIFIED-Z SCORE!! 
#Counts$Z_Score <- ((Counts$Corrected_BScore - mean(Counts$Corrected_BScore)) /mad(Counts$Corrected_BScore))

CountsX <- gene_template[which(gene_template$Z_Score > 2),]

#Filter for only targets predicted to interact directly with targets (no results for Zinc)
#Counts <- Counts[which(Counts$Interactor == "triterpene" | Counts$Interactor == "disulfiram" | Counts$Interactor == "plumbagin",),]
#For chemical screen
#genes_to_test <- Counts[[1]]
#For CRISPR Screen
genes_to_test <- CountsX[[3]]

# Example enrichGO analysis
enriched <- enrichGO(
    gene = genes_to_test,
    OrgDb = org.Hs.eg.db,
    keyType = "SYMBOL",
    ont = "BP",  # Biological Process
   # pAdjustMethod = "BH",
    pvalueCutoff = 0.05
)

# Extract GO results

go_results <- as.data.frame(enriched[1:20])

go_results

#go_results <- go_results[which(go_results$qvalue < 0.1),]

# Create a gene-term matrix based on p-value

genes <- unique(unlist(strsplit(go_results$geneID, "/")))
go_terms <- go_results$Description


#go_terms3 <- go_terms3[1:20]
# Initialize matrix for p-values
gene_term_matrix <- matrix(NA, nrow = length(go_terms), ncol = length(genes),
                           dimnames = list(go_terms, genes))

# Populate matrix with -log10(p-value) for visualization
for (i in seq_along(go_terms)) {
    term_genes <- unlist(strsplit(go_results$geneID[i], "/"))
    p_value <- go_results$p.adjust[i]
    gene_term_matrix[go_terms[i], term_genes] <- -log10(p_value)  # Use -log10(p-value)
}

# Replace NA with 0 for better visualization
gene_term_matrix[is.na(gene_term_matrix)] <- 0

#For Genomic Screen
annotation <- data.frame(Z_Score = Counts$Z_Score[match(colnames(gene_term_matrix), Counts[[3]])])

rownames(annotation) <- colnames(gene_term_matrix)


phmap <- pheatmap(
  
    gene_term_matrix,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = TRUE,
    show_colnames = TRUE,
    
    annotation_col = annotation,


    color = colorRampPalette(c("gray92", "darkred"))(50),
    border_color = "black",
    fontsize = 16,               # General font size
  fontsize_row = 18,           # Font size for row labels
  fontsize_col = 18,
  cutree_rows = 8, cutree_cols = 27,

)

ggsave("/Users/dylan/Documents/papers_grants_other/SmallMoleculeNuclearCondensates/GO_BPHeatMapWTScreen.png", phmap, dpi = 300, width = 30, height = 16)

library(heatmaply)

# Create the heatmaply object
heatmap_plot <- heatmaply(
  (gene_term_matrix), 
  Rowv = phmap[[1]], #To keep the rows/columns ordered in the same arrangement to that of phmap
  Colv = phmap[[2]], 
  #k_col = 2, 
  k_row = 3, 
  fontsize_row = 5, 
  fontsize_col = 6, 
  col_side_colors = annotation, 
  xlab = NULL, 
  ylab = NULL, 
  margins = c(1, 1, 1, 1),
  plot_width = 2000, 
  plot_height = 800, 
  color = colorRampPalette(c("gray92", "darkred"))(50),

  file = "/Users/dylan/Documents/CRISPR Screen data/WT/InteractiveHeatmap.html",

)

# Plot the top 30 categories
go_results


GOPlot1 <- plot(barplot(enriched, showCategory = 20, title = "Top 20 GO Terms - Genetic Screen:"), )+  theme(plot.title = element_text(size = 20)) 

ggsave("/Users/dylan/Documents/CRISPR Screen data/WT/GO_barPlotWTScreen.png", GOPlot1, dpi = 300, width = 10, height = 15)

genes
```


```{r}

Counts2 <- read.delim("/Users/dylan/Documents/CRISPR Screen data/WT/StitchResults.csv", 
                     header = TRUE, sep = ",")

#Filter for only targets predicted to interact directly with targets (no results for Zinc)
Counts2 <- Counts2[which(Counts2$Interactor == "triterpene" | Counts2$Interactor == "disulfiram" | Counts2$Interactor == "plumbagin",),]
#For chemical screen
genes_to_test2 <- Counts2[[1]]

# Example enrichGO analysis
enriched2 <- enrichGO(
    gene = genes_to_test2,
    OrgDb = org.Hs.eg.db,
    keyType = "SYMBOL",
    ont = "BP",  # Biological Process
    pAdjustMethod = "BH",
    qvalueCutoff = 0.05
)

# Extract GO results


go_results2 <- as.data.frame(enriched2[1:20])
go_results2

# Create a gene-term matrix based on p-value

genes2 <- unique(unlist(strsplit(go_results2$geneID, "/")))
go_terms2 <- go_results2$Description

go_terms2 <- go_terms2[1:20]
# Initialize matrix for p-values
gene_term_matrix2 <- matrix(NA, nrow = length(go_terms2), ncol = length(genes2),
                           dimnames = list(go_terms2, genes2))

# Populate matrix with -log10(p-value) for visualization
for (i in seq_along(go_terms2)) {
    term_genes <- unlist(strsplit(go_results2$geneID[i], "/"))
    p_value <- go_results2$p.adjust[i]
    gene_term_matrix2[go_terms2[i], term_genes] <- -log10(p_value)  # Use -log10(p-value)
}



# Replace NA with 0 for better visualization
gene_term_matrix2[is.na(gene_term_matrix2)] <- 0

Counts2 <- Counts2 %>%
    mutate_all(funs(str_replace(., "disulfiram", "Disulfiram"))) %>%
    mutate_all(funs(str_replace(., "triterpene", "Celastrol"))) %>%
    mutate_all(funs(str_replace(., "plumbagin", "Plumbagin")))

#For chemical screen
annotation2 <- data.frame(Drug = Counts2$Interactor[match(colnames(gene_term_matrix2), Counts2[[1]])])
rownames(annotation2) <- colnames(gene_term_matrix2)

# phmap2 <- pheatmap(
#   
#     gene_term_matrix2,
#     cluster_rows = TRUE,
#     cluster_cols = TRUE,
#     show_rownames = TRUE,
#     show_colnames = TRUE,
#     main = "Enriched GO Biological Process from Predicted Targets of Chemical Screen",
#     
#     annotation_col = annotation2,
# 
#     color = colorRampPalette(c("white", "red"))(50),
#     border_color = NA,
#     fontsize = 14,               # General font size
#   fontsize_row = 15,           # Font size for row labels
#   fontsize_col = 13,      
# )
# 
# 
# phmap2 + 
#   theme(
#     plot.title = element_text(size = 18, face = "bold"),
#     legend.text = element_text(15),
#     theme(title = element_text(size = 32, face = "bold")) +
#     theme(axis.title.x = element_text(size = 60, face = "bold")))

library(grid)

phmap2 <- pheatmap(
  
    gene_term_matrix2,
    cluster_rows = TRUE,
    cluster_cols = TRUE,
    show_rownames = TRUE,
    show_colnames = TRUE,
    main = " " ,
    
    annotation_col = annotation2,

    color = colorRampPalette(c("gray92", "darkred"))(50),
    #border_color = "red",
    fontsize = 16,               # General font size
  fontsize_row = 22,           # Font size for row labels
  fontsize_col = 24,
  cutree_rows = 10, cutree_cols = 10,

)

# 
# phmap2 +
#   theme(
#     plot.title = element_text(size = 18, face = "bold"),
#     legend.text = element_text(12, face = "bold"),
#     theme(title = element_text(size = 32, face = "bold")) +
#     theme(axis.title.x = element_text(size = 60, face = "bold")))


ggsave("/Users/dylan/Documents/papers_grants_other/SmallMoleculeNuclearCondensates/GO_BPHeatMapPharmakonScreen2.png", phmap2, dpi = 300, width = 20, height = 24)


library(heatmaply)

# Create the heatmaply object
heatmap_plot <- heatmaply(
  (gene_term_matrix2), 
  Rowv = phmap2[[1]], #To keep the rows/columns ordered in the same arrangement to that of phmap
  Colv = phmap2[[2]], 
  #k_col = 2, 
  k_row = 3, 
  fontsize_row = 8, 
  fontsize_col = 10, 
  col_side_colors = annotation2, 
  xlab = NULL, 
  ylab = NULL, 
  margins = c(1, 1, 1, 1),
  plot_width = 2000, 
  plot_height = 800, 
  color = colorRampPalette(c("gray92", "darkred"))(50),

  file = "/Users/dylan/Documents/papers_grants_other/SmallMoleculeNuclearCondensates/PharmakonInteractiveHeatmap.html",

)





go_results2 <- as.data.frame(enriched2@result)
go_results2 <- as.data.frame(enriched2[1:20])

# Plot the top 30 categories
go_results2

GOPlot2 <- plot(barplot(enriched2, showCategory = 20, title = "Top 20 GO Terms - Chemical Screen:")) +  theme(plot.title = element_text(size = 20)) 

ggsave("/Users/dylan/Documents/CRISPR Screen data/WT/GO_barPlotPharmakonScreen.png", GOPlot2, dpi = 300, width = 10, height = 14)


go_terms
go_terms2

genes
genes2

for (i in 1:length(go_terms2)){
  if(go_terms2[i]  %in% go_terms){
    print(go_terms2[i])
  }
}



for (i in 1:length(genes2)){
  if(genes2[i]  %in% genes){
    print(genes2[i])
  }
}

# 
# Genes overlapping:
# [1] "SRC"
# [1] "HSP90AA1"
# 
# BP overlapping:
# [1] "response to heat"
# [1] "response to temperature stimulus"
# [1] "response to reactive oxygen species"

#GeneRatio (k/n): "ratio of input genes that are annotated in a term." (# of genes in the input list that are annotated to the corresponding term (function) / # of genes in input list).

#BgRatio (M/N): "ratio of all genes that are annotated in this term" (The number of genes that are annotated to the term / The total number of genes in the background gene list #("universe")).

```

```{r}

# Identify xunique gene names from both matrices
genes_matrix1 <- colnames(gene_term_matrix)
genes_matrix2 <- colnames(gene_term_matrix2)

genes_matrix1 <- colnames(gene_term_matrix)
genes_matrix2 <- colnames(gene_term_matrix2)

venn_data <- list(
    "Genetic Screen Hits" = genes_matrix1,
     "Predicted Targets from Chemical Screen" = genes_matrix2
)
library(ggvenn)
# Create the Venn diagram
ggvenn_plot <- ggvenn(
    venn_data, 
    fill_color = c("darkred", "green"),
    stroke_size = 0.5,
    set_name_size = 5,
    show_percentage = TRUE,
    text_size = 7 # This will show the percentages inside the Venn diagram
)

# Extract the overlap genes
overlap_genes <- intersect(genes_matrix1, genes_matrix2)


genes_matrixPath1 <- rownames(gene_term_matrix)
genes_matrixPath2 <- rownames(gene_term_matrix2)

venn_data_pathways <- list(
    "Enriched GO Pathways (Genetic Screen)" = genes_matrixPath1,
    "Enriched GO Pathways (Chemical Screen)" = genes_matrixPath2
)

# Create the Venn diagram
ggvenn_plot_Path <- ggvenn(
    venn_data_pathways, 
    fill_color = c("darkred", "green"),
    stroke_size = 0.5,
    set_name_size = 5,
    show_percentage = TRUE,
    text_size = 7 
    # This will show the percentages inside the Venn diagram
) 

# Extract the overlap genes 
overlap_genes <- intersect(genes_matrixPath1, genes_matrixPath2)


ggsave("/Users/dylan/Documents/CRISPR Screen data/WT/VennGeneOverlap.png", ggvenn_plot, dpi = 300, width = 10, height = 15)
ggsave("/Users/dylan/Documents/CRISPR Screen data/WT/VennPathOverlap.png", ggvenn_plot_Path, dpi = 300, width = 10, height = 15)

```


```{r}


# Identify unique gene names from both matrices
genes_matrix1 <- colnames(gene_term_matrix)
genes_matrix2 <- colnames(gene_term_matrix2)

# Identify overlapping and unique genes
overlap_genes <- intersect(genes_matrix1, genes_matrix2)
unique_matrix1 <- setdiff(genes_matrix1, overlap_genes)
unique_matrix2 <- setdiff(genes_matrix2, overlap_genes)

combined_genes <- c(genes_matrix1, genes_matrix2)

combined_genes <- unique(combined_genes)


allgenes <- read.delim("/Users/dylan/Documents/CRISPR Screen data/WT/WTRawData.csv", 
                     header = TRUE, sep = ",")

bscore_vector2 <- allgenes$BScore[match(genes_matrix2, allgenes$Gene)]



# Step 4: Fill the matrix with BScore values
# Now, populate the matrix with BScore for each gene in both rows and columns
for (i in seq_along(combined_genes)) {
    gene <- combined_genes[i]
    gene_term_matrix3[gene, ] <- bscore_vector[i]
    gene_term_matrix3[, gene] <- bscore_vector[i]
}

# The matrix is now filled with the BScore values corresponding to each gene
print(gene_term_matrix3)
gene_term_matrix3 <- matrix(NA, nrow = length(combined_genes), ncol = length(combined_genes),
                           dimnames = list(combined_genes, combined_genes))





correlation_matrix[is.na(correlation_matrix)] <- 0

correlation_matrix <- cor(t(gene_term_matrix3), method = "pearson")



pheatmap(
  gene_term_matrix3,
  main = "Correlation Between Gene-Term Matrices",
  color = colorRampPalette(c("blue", "white", "red"))(50),
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  border_color = NA
)



Counts <- read.delim("/Users/dylan/Documents/CRISPR Screen data/WT/WTRawData.csv", 
                     header = TRUE, sep = ",")
#Counts <- Counts[which(Counts$Z_Score > 3),]

Counts


Counts$Z_Score <- ((Counts$BScore - mean(Counts$BScore)) /sd(Counts$BScore))
  
  
merged_Df_NoCtrls_noNA <- merged_Df_NoCtrls[complete.cases(merged_Df_NoCtrls),]
merged_Df_NoCtrls$Z_Score <-  ((merged_Df_NoCtrls$Corrected_BScore - mean(merged_Df_NoCtrls_noNA$Corrected_BScore)) / sd(merged_Df_NoCtrls_noNA$Corrected_BScore))
merged_Df_NoCtrls$Z_Score <- round(merged_Df_NoCtrls$Z_Score, 2)


mad(Counts$Corrected_BScore)
sd(Counts$BScore)
```


```{r}

# 
# #Genome Screen
# gene_template <- read.delim("/Users/dylan/Documents/CRISPR Screen data/WT/WTRawData.csv", 
#                      header = TRUE, sep = ",")
# gene_template <- Counts
# normalized_Sd_Bar <- ggplot(data = gene_template, aes(x=paste0(Well_Condition, Plate), y = Z_Score, fill = factor(ifelse(Z_Score >= 3, "tofill", "nottofill")))) +
#   geom_point(data = filter(gene_template, Z_Score >= 4),
#                aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = Z_Score , yend = Z_Score),
#                color = "red", size = 2) +
#   geom_point(data = filter(gene_template, Z_Score < 4),
#                aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = Z_Score, yend = Z_Score),
#                color = "black", size =0.1, position = position_jitter(width = 0.3, height = 0), alpha = 0.5)+
#   geom_point(data = filter(gene_template, Gene == "TOR1A" | Gene == "TOR1B"),
#                aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = Z_Score, yend = Z_Score),
#                color = "green", size = 3 )+
#    # geom_segment(data = filter(merged_Df_NoCtrls, Gene == "MLF2"),
#    #             aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = 0, yend = Z_Score),
#    #             color = "purple", size =1 )+
#    # geom_segment(data = filter(merged_Df_NoCtrls, Gene == "OR5D18" | Gene == "POLR2I" |Gene ==  "POLR2A"),
#    #             aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = 0, yend = Z_Score),
#    #             color = "gray", size =1 )+
#    
#       scale_fill_manual(values = c("tofill" = "red", "nottofill" ="gray27")) +
# 
#    theme_classic() +
#    theme(axis.title.x=element_blank(),
#          axis.text.x=element_blank(),
#          axis.ticks.x=element_blank(),) +
# # #Edit title as needed
# # #------------------------------------------------------------------
#    labs(y = "Z-Score", x = "sgRNA", title = "Genome-Wide CRISPR Screen Hits:") +
# # #------------------------------------------------------------------
# #
#   scale_y_continuous(limits = c(-10, -5, -3, 0, 3, 5, 10)) +
#   scale_y_continuous(breaks=c( -10, -8, -6, -4, -2, 0, 2, 4,6,8,10)) +
#   geom_hline(yintercept = c(2), color = "darkred", linetype = "dashed", size = 0.5) +
#   geom_text_repel(data = filter(gene_template, Z_Score >= 4 |Gene == "TOR1A" | Gene == "TOR1B"  ), aes(label = paste0(Gene)), box.padding = 0.2, size = 3, max.overlaps = Inf) +
#   guides(fill="none")+
#   theme(plot.title = element_text(size = 18)) +
#   theme(axis.title = element_text(size = 18)) +
#   coord_cartesian(ylim = c(-10, 10))
# 
# normalized_Sd_Bar
# ggsave("/Users/dylan/Documents/CRISPR Screen data/WT/WTNormalizedZScore.png", normalized_Sd_Bar, dpi = 300, width = 10, height = 10)
# 

head(gene_template)
```


```{r}
library(readxl)
library(ggrepel)
#Pharmakon Screen
gene_template <- read_excel("/Users/dylan/Documents/Analyzed_Data_dp/DP/Pharmakon_Screen/Pharmakon_2h_1.xlsx", sheet = "Raw_Data")

#USE MODIFIED-Z SCORE!! 
#gene_template$Z_Score <- ((gene_template$Corrected_BScore - mean(gene_template$Corrected_BScore)) / mad(gene_template$Corrected_BScore))
gene_template <- gene_template[gene_template$Compound != "NA",]

normalized_Sd_Bar <- ggplot(data = gene_template, aes(x=paste0(Well_Condition, Plate), y = Z_Score, fill = factor(ifelse(Z_Score >= 3, "tofill", "nottofill")))) +
  geom_point(data = filter(gene_template, Z_Score >= 3),
               aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = Z_Score , yend = Z_Score),
               color = "blue", size = 2.25) +
  geom_point(data = filter(gene_template, Z_Score < 3),
               aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = Z_Score, yend = Z_Score),
               color = "black", size =1, position = position_jitter(width = 0.3, height = 0), alpha = 0.4)+
  geom_point(data = filter(gene_template, Compound == "PLUMBAGIN" | Compound == "CELASTROL" | Compound == "PYRITHIONE ZINC" | Compound == "DISULFIRAM"),
               aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = Z_Score, yend = Z_Score),
               color = "red", size = 4 )+
  
   # geom_segment(data = filter(merged_Df_NoCtrls, Gene == "MLF2"),
   #             aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = 0, yend = Z_Score),
   #             color = "purple", size =1 )+
   # geom_segment(data = filter(merged_Df_NoCtrls, Gene == "OR5D18" | Gene == "POLR2I" |Gene ==  "POLR2A"),
   #             aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = 0, yend = Z_Score),
   #             color = "gray", size =1 )+
   
      scale_fill_manual(values = c("tofill" = "red", "nottofill" ="black")) +

   theme_classic() +
   theme(axis.title.x=element_blank(),
         axis.text.x=element_blank(),
         axis.ticks.x=element_blank(),
          axis.text.y = element_text(size = 18, face = "bold"), # Increase y-axis label size
    axis.title.y = element_text(size = 24, face = "bold") # Adjust axis title size if needed
  ) +
# #Edit title as needed
# #------------------------------------------------------------------
   labs(y = "Z-Score", x = "Drugs", title = NULL) +
# #------------------------------------------------------------------
#
  scale_y_continuous(limits = c(-10, -5, -3, 0, 3, 5, 8)) +
  scale_y_continuous(breaks=c( -10, -8, -6, -4, -2, 0, 2, 4,6,8)) +
  geom_hline(yintercept = c(3), color = "darkred", linetype = "dashed", size = 0.5) +
  geom_text_repel(data = filter(gene_template, Z_Score >= 3 & Compound != "PLUMBAGIN" & Compound != "CELASTROL" & Compound != "PYRITHIONE ZINC" & Compound != "DISULFIRAM"), aes(label = paste0(Compound)), box.padding = 0.5, size = 4, max.overlaps = Inf, fontface = "bold") +
  geom_text_repel(data = filter(gene_template, Compound == "PLUMBAGIN" | Compound == "CELASTROL"| Compound == "PYRITHIONE ZINC" | Compound == "DISULFIRAM"), aes(label = paste0(Compound)), box.padding = 2, size = 5, max.overlaps = Inf, fontface = "bold", color = "red") +

  
  guides(fill="none")+
  
  theme(plot.title = element_text(size = 18)) +
  theme(axis.title = element_text(size = 24, face = "bold")) +
  coord_cartesian(ylim = c(-2.5, 9))

normalized_Sd_Bar
ggsave("/Users/dylan/Documents/Analyzed_Data_dp/DP/Pharmakon_Screen/Pharmakon17602hV2.tif", normalized_Sd_Bar, dpi = 300, width = 10, height = 10)
ggsave("/Users/dylan/Documents/papers_grants_other/SmallMoleculeNuclearCondensates/Pharmakon1760ZScore.pdf", normalized_Sd_Bar, dpi = 300, width = 10, height = 10)

head(gene_template)
```

```{r}
# save_Location <- "C:/Users/Schlieker Lab/Desktop/Old_Backedup_Data_prior_to_23_11_17/Dylan/Compound Screen/pilot_screen_9_1_23_dp/pilot_6h_results.xlsx"
# wb <- loadWorkbook(gene_template)
 
merged_Df_NoCtrls <- gene_template
merged_Df_NoCtrls[is.na(merged_Df_NoCtrls$Plate) == FALSE,]


#FOR Pharmakon ANALYSIS
raw_Distribution <- ggplot(merged_Df_NoCtrls, aes(x = Average_foci_perCell )) +
  geom_histogram(color = "black", fill = "darkred", binwidth = 0.3) +
  geom_density(alpha = 0.6, fill = "darkred" ) +
  labs(title = NULL,  y = "Count", x = "MLF2-GFP Foci/Cell") +
  theme_classic() +
  theme(title = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 37, face = "bold")) +
  theme(axis.text.y = element_text(size = 30, face = "bold"), # Increase y-axis label size
    axis.text.x = element_text(size = 30, face = "bold"),
     axis.ticks.length = unit(0.4, "cm"), # Increase tick length
    axis.ticks = element_line(size = 1.2) # Increase tick thickness# Adjust axis title size if needed
  ) 


#FOR Pharmakon
normalized_Dist <- ggplot(merged_Df_NoCtrls, aes(x = Normalized_Effect)) +
  geom_histogram(color = "black", fill = "darkred", binwidth = 1) +
  geom_density(alpha = 0.6, fill = "darkgray" ) +

  geom_vline(aes(xintercept = 100), linetype = 'dashed', size = 2) +
  annotate("text", x=90, y=50, label="WT HeLa", angle=90, size = 12) +
  geom_vline(aes(xintercept = 0), linetype = 'dashed', size = 2) +
  annotate("text", x=-10, y=50, label= "DMSO",  color = "black", angle=90, size = 12) +


  # geom_vline(aes(xintercept = mean_minus_3SD), linetype = 'dashed', color = "red", size = 1) +
  # annotate("text", x=mean_minus_3SD - 4, y=25, label="(-) 3SD", angle=90, size = 5) +

  labs(title = NULL, y = "Count", x = paste("% Decrease in MLF2-GFP Relative to DMSO")) +
  theme_classic() +
   theme(title = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 37, face = "bold")) +
  theme(axis.text.y = element_text(size = 30, face = "bold"), # Increase y-axis label size
    axis.text.x = element_text(size = 30, face = "bold"),
     axis.ticks.length = unit(0.4, "cm"), # Increase tick length
    axis.ticks = element_line(size = 1.2) # Increase tick thickness# Adjust axis title size if needed
  ) 

library(ggpubr)
image_Save_Location <- "/Users/dylan/Documents/Analyzed_Data_dp/DP/Pharmakon_Screen/"
# #Exports graphs as TIFF files
normalized_Dist
tiff(paste(image_Save_Location, "distrpilot_", ".tiff" ,sep = ""), units="in", width=12, height=10, res=300)
ggarrange(normalized_Dist, heights = c(6, 6))
dev.off()

raw_Distribution
tiff(paste(image_Save_Location, "rawdistrpilot_", ".tiff" ,sep = ""), units="in", width=12, height=10, res=300)
ggarrange(raw_Distribution, heights = c(6, 6))
dev.off()

```

```{r}

library(readxl)
library(ggrepel)
gene_template <- read_excel("D:/CRISPR_Screen_Analysis_dp/DiversityChemicalScreen/results/LifeChemicalsDiv_AllRuns_YCMD_Names.xlsx")
colnames(gene_template)[5] <- "Gene"
colnames(gene_template)[17] <- "Z_Score"

merged_Df_NoCtrls <- gene_template
 
normalized_Sd_Bar <- ggplot(data = gene_template, aes(x=SMILES, y = Z_Score, fill = factor(ifelse(Z_Score >= 30, "tofill", "nottofill")))) +
  geom_point(data = filter(gene_template, Z_Score >= 30),
               aes(x = SMILES, xend =SMILES, y = Z_Score , yend = Z_Score),
               color = "red", size = 2.25) +
  geom_point(data = filter(gene_template, Z_Score < 30),
               aes(x = SMILES, xend = SMILES, y = Z_Score, yend = Z_Score),
               color = "black", size =1, position = position_jitter(width = 0.3, height = 0), alpha = 0.25)+
  # geom_point(data = filter(gene_template, Gene == "TOR1A" | Gene == "TOR1B"),
  #              aes(x = Gene, xend = Gene, y = Z_Score, yend = Z_Score),
  #              color = "green", size = 5 )+
  # geom_point(data = filter(gene_template, Gene ==  "APOL3" | Gene == "ZNF335" | Gene == "RNF26"  | Gene == "RAN"),
  #              aes(x = Gene, xend = Gene, y = Z_Score, yend = Z_Score),
  #              color = "red", size = 4 )+
  
  
   # geom_segment(data = filter(merged_Df_NoCtrls, Gene == "MLF2"),
   #             aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = 0, yend = Z_Score),
   #             color = "purple", size =1 )+
   # geom_segment(data = filter(merged_Df_NoCtrls, Gene == "OR5D18" | Gene == "POLR2I" |Gene ==  "POLR2A"),
   #             aes(x = paste0(Well_Condition, Plate), xend = paste0(Well_Condition, Plate), y = 0, yend = Z_Score),
   #             color = "gray", size =1 )+
   
      scale_fill_manual(values = c("tofill" = "red", "nottofill" ="black")) +

   theme_classic() +
   theme(axis.title.x=element_blank(),
         axis.text.x=element_blank(),
         axis.ticks.x=element_blank(),
          axis.text.y = element_text(size = 18, face = "bold"), # Increase y-axis label size
    axis.title.y = element_text(size = 24, face = "bold") # Adjust axis title size if needed
  ) +
# #Edit title as needed
# #------------------------------------------------------------------
   labs(y = "%Normalized Effect", x = "Candidate Genes", title = "Life Chemicals Diversity, >30k Novel Compounds:") +
# #------------------------------------------------------------------
#
  scale_y_continuous(
  limits = c(0, 75),
  breaks = c( 0, 25, 50, 75)
) +
  geom_hline(yintercept = c(30), color = "darkred", linetype = "dashed", size = 1) +
  
  geom_text_repel(data = filter(gene_template, Z_Score >= 35 & Gene != "TOR1A" & Gene != "TOR1B" & Gene !=  "APOL3" & Gene != "ZNF335" & Gene != "RNF26"  & Gene != "RAN" ), aes(label = paste0(Gene)), box.padding = 0.4, size = 3.25, max.overlaps = Inf, fontface = "bold") +
  
  geom_text_repel(data = filter(gene_template, Gene == "TOR1A" | Gene == "TOR1B" |Gene ==  "APOL3" | Gene == "ZNF335" | Gene == "RNF26" | Gene == "RAN" ), aes(label = paste0(Gene)), box.padding = 3.5, size = 8, max.overlaps = Inf, fontface = "bold", color = "red") +


  guides(fill="none")+
  theme(plot.title = element_text(size = 24, face = "bold")) +
  theme(axis.title = element_text(size = 24, face = "bold")) +
  coord_cartesian(ylim = c(0.3, 75))

normalized_Sd_Bar
ggsave("D:/CRISPR_Screen_Analysis_dp/DiversityChemicalScreen/results/NormalizedScore.tiff", normalized_Sd_Bar, dpi = 300, width = 10, height = 10)

head(gene_template)


```


```{r}
#WT Screen Distributions:

library(readxl)
library(ggrepel)
gene_template <- read_excel("D:/CRISPR_Screen_Analysis_dp/DiversityChemicalScreen/results/LifeChemicalsDiv_AllRuns_YCMD_Names.xlsx")


merged_Df_NoCtrls <- gene_template
#merged_Df_NoCtrls[is.na(merged_Df_NoCtrls$Plate) == FALSE,]


#FOR Pharmakon ANALYSIS
raw_Distribution <- ggplot(merged_Df_NoCtrls, aes(x = Average_foci_perCell )) +
  geom_histogram(color = "black", fill = "darkred", binwidth = 0.05) +
  geom_density(alpha = 0.6, fill = "darkred" ) +
  labs(title = NULL,  y = "Count", x = "MLF2-GFP Foci/Cell") +
  theme_classic() +
  theme(title = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 37, face = "bold")) +
  theme(axis.text.y = element_text(size = 30, face = "bold"), # Increase y-axis label size
    axis.text.x = element_text(size = 40, face = "bold"),
     axis.ticks.length = unit(0.4, "cm"), # Increase tick length
    axis.ticks = element_line(size = 1.2) # Increase tick thickness# Adjust axis title size if needed
  ) 


#FOR Pharmakon
normalized_Dist <- ggplot(merged_Df_NoCtrls, aes(x = Normalized_Effect)) +
  geom_histogram(color = "black", fill = "darkred", binwidth = 0.5) +
  geom_density(alpha = 0.6, fill = "darkgray" ) +

  geom_vline(aes(xintercept = 100), linetype = 'dashed', size = 2) +
  annotate("text", x=90, y=600, label="Tor1A/Tor1B sgRNA", angle=90, size = 12) +
  geom_vline(aes(xintercept = 0), linetype = 'dashed', size = 2) +
  annotate("text", x=-10, y=600, label= "NT sgRNA Ctrl",  color = "black", angle=90, size = 12) +


  # geom_vline(aes(xintercept = mean_minus_3SD), linetype = 'dashed', color = "red", size = 1) +
  # annotate("text", x=mean_minus_3SD - 4, y=25, label="(-) 3SD", angle=90, size = 5) +

  labs(title = NULL, y = "Count", x = paste("% Increase in MLF2-GFP Relative to NT")) +
  theme_classic() +
   theme(title = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 37, face = "bold")) +
  theme(axis.text.y = element_text(size = 30, face = "bold"), # Increase y-axis label size
    axis.text.x = element_text(size = 30, face = "bold"),
     axis.ticks.length = unit(0.4, "cm"), # Increase tick length
    axis.ticks = element_line(size = 1.2) # Increase tick thickness# Adjust axis title size if needed
  ) 

image_Save_Location <- "/Users/dylan/Documents/papers_grants_other/SmallMoleculeNuclearCondensates/"
# #Exports graphs as TIFF files
normalized_Dist
tiff(paste(image_Save_Location, "distrpilot_WTCRISPR", ".tiff" ,sep = ""), units="in", width=12, height=10, res=300)
ggarrange(normalized_Dist, heights = c(6, 6))
dev.off()

raw_Distribution
tiff(paste(image_Save_Location, "rawdistrpilot_WTCRISPR", ".tiff" ,sep = ""), units="in", width=12, height=10, res=300)
ggarrange(raw_Distribution, heights = c(6, 6))
dev.off()


```




```{r}
#For the tighter single-cell-level dose responses: 

library(openxlsx)
library(ggplot2)
library(data.table)
library(tidyr)
library(ggpval)
library(ggprism)
library(patchwork)
library(magrittr)
library(dplyr)
library(tidyverse)
library(rstatix)
library(ggpubr)
library(RColorBrewer)

wb <- createWorkbook()
addWorksheet(wb, "original_df")
addWorksheet(wb, "summary_info")
setwd("C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/compound_screen/Pharmakon/export_tighterdoseresponse_8h3h/")

pathName <- "C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/compound_screen/Pharmakon/export_tighterdoseresponse_8h3h/"
fileName <- "MLF2_GFPnuclear_periphery.csv"
  cyto_filename= paste0(pathName, fileName)
  
  df1 <- read.csv(cyto_filename)
  df1
  
  #Creates a subset of the dataframe with the relevant columns. May need to adjust if your analysis included/excluded some other data.
  sub_df <- df1[, c(1,2,3,7)] #DsRed, GFP
  colnames(sub_df)[3] <- "Names"
  colnames(sub_df)[4] <- "GFP_Foci"
  
   replace_Names <- function(x) {
    
    x <- gsub(".*celastrol_lowest.*", "Celastrol_-2x", x)
    x <- gsub(".*celastrol_evenlower.*", "Celastrol_IC50", x)
    x <- gsub(".*celastrol_lesshigh.*", "Celastrol_2x", x)
    x <- gsub(".*celastrol_high.*", "Celastrol_8x", x)
    

    x <- gsub(".*dis_lowest.*", "Disulfiram_-2x", x)
    x <- gsub(".*dis_evenlower.*", "Disulfiram_IC50", x)
    x <- gsub(".*dis_lesshigh.*", "Disulfiram_2x", x)
    x <- gsub(".*dis_high.*", "Disulfiram_8x", x)
    
    x <- gsub(".*plumbagin_lowest.*", "Plumbagin_-2x", x)
    x <- gsub(".*plumbagin_evenlower.*", "Plumbagin_IC50", x)
    x <- gsub(".*plumbagin_lesshigh.*", "Plumbagin_2x", x)
    x <- gsub(".*plumbagin_high.*", "Plumbagin_8x", x)
    
    x <- gsub(".*pyrithioneZinc_lowest.*", "Pyrithione Zinc_-2x", x)
    x <- gsub(".*pyrithioneZinc_evenlower.*", "Pyrithione Zinc_IC50", x)
    x <- gsub(".*pyrithioneZinc_lesshigh.*", "Pyrithione Zinc_2x", x)
    x <- gsub(".*pyrithioneZinc_high.*", "Pyrithione Zinc_8x", x)
    
    x <- gsub(".*PO_.*", NA, x)
    x <- gsub(".*DMSO.*", "DMSO_DMSO", x)
    

    
  }
  
sub_df

  sub_df2 <- sub_df
  sub_df2$Names <- sapply(sub_df2$Names,replace_Names)
  sub_df2 <- drop_na(sub_df2)
  
  unique(sub_df2$Names)
  
  #MLF2_Foci_Sheet <- addWorksheet(wb, sheetName = variable_of_interest)
  
    
  #Separate names into compound name, time, and concentration
  sub_df2 <- sub_df2 %>% separate(Names, into = c("Compound",  "Concentration"), sep = "_")

  #Define levels for bar placement:
  sub_df2$Compound <- factor(sub_df2$Compound, levels = c("DMSO",  "Pyrithione Zinc", "Celastrol", "Plumbagin", "Disulfiram"))
  sub_df2$Concentration <- factor(sub_df2$Concentration, levels = c("DMSO", "-2x",  "IC50", "2x", "8x"))

 
library(viridis)           

jittered_BoxPlot <-  ggplot(sub_df2, aes(x = Compound,  y = GFP_Foci, fill = Concentration)) + 
    # stat_summary(data = stats_Summary,   geom = "text", aes(y = .data[[variable_of_Interests]],label = round(..y.., 2)), size = 7, vjust = -0.25, hjust = 2, col = "red",
    #              position = position_dodge(0.75), show.legend = FALSE, color = "black") +

    geom_jitter(position = position_jitterdodge(jitter.width = 0.1, dodge.width = 0.75), size = 2, show.legend = TRUE) +
    
    geom_boxplot(alpha = 0.5, width = 0.75) +
    # geom_errorbar(data = stats_Summary, mapping = aes(x = Names, ymin = .data[[variable_of_Interests]] - .data[[variable_2]], ymax = .data[[variable_of_Interests]] + .data[[variable_2]], width= 0.1, size  = 1), color = "black", size = 1)+
    # geom_point(data = stats_Summary, aes(y = .data[[variable_of_Interests]]), geom = "point", size = 3, color = "black", show.legend = FALSE) +
    
    theme_classic() +
    labs(y = "MLF2-GFP Foci/Cell", title = NULL, fill = "Concentration:") + 
    theme(plot.title = element_text(size = 20, face = "bold", hjust = 0.5)) +
    theme(axis.title.y = element_text(size = 30, face = "bold"), axis.title.x = element_blank(), axis.ticks.length = unit(0.1, "cm"), axis.text.y = element_text(size = 22, face = "bold")) +
      theme(axis.text.x = element_text(size = 24, angle = 55, hjust = 1, face = "bold")) +
    theme(legend.text = element_text(size = 24, face = "bold"),
          legend.title = element_text(size = 24, face = "bold")) +
          theme(legend.position = c(0.85, 0.8), legend.background = element_blank()) + # Adjusted for vertical centering
    


    coord_cartesian(ylim = c(0, 450)) +
  scale_y_continuous(breaks = c(0,100,200,300, 400)) +
  scale_fill_viridis(discrete = TRUE, option = "B", direction =  -1,  begin = 0, end = 0.75) 



jittered_BoxPlot

       
tiff("C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/grants_drafts/poch_2024/tighterdoseresponse_8hDox_3hCompound.tif", units="in", width=10, height=10, res=300)
ggarrange(jittered_BoxPlot, heights = c(10, 10))
dev.off()

ggsave(ggsave("C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/grants_drafts/poch_2024/tighterdoseresponse_8hDox_3hCompound.PDF"), jittered_BoxPlot, dpi = 300, width = 10, height = 10)
```

```{r}
#For the K48/MLF2-GFP final concentration boxplots:
library(ggpval)
library(openxlsx)
library(ggplot2)
library(EnvStats)
library(ggpubr)
library(psych)
library(stringr)
library(dplyr)
library(plotrix)
library(tidyr)
library(viridis)           
library(ggpubr)
library(ggsignif)
library(rstatix)
library(scales)


#Create new Excel Sheet
wb <- createWorkbook()
graph_Analysis <- addWorksheet(wb, sheetName ='Analysis')

#Sets the working directory to the location your spreadsheet is at.
#setwd("C:/Users/dylan/OneDrive/Yale/Schlieker_Lab/cell_profiler_rawdata/CellProfiler_exp_Image.csv")
main_foci_file = "C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/compound_screen/Pharmakon/finalConcentrations_export/Analysis/MultipleMarkersnuclear_periphery.csv"
nuc_file = "C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/compound_screen/Pharmakon/finalConcentrations_export/Analysis/MultipleMarkersNuclei.csv"

main_foci_file2 = "C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/compound_screen/Pharmakon/finalConcentrations_export/Analysis_Batch2/MultipleMarkersnuclear_periphery2.csv"
nuc_file2 = "C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/compound_screen/Pharmakon/finalConcentrations_export/Analysis_Batch2/MultipleMarkersNuclei2.csv"

df_nuclear_Foci <- read.csv(main_foci_file)
df_nuclear_Size <- read.csv(nuc_file)
df_nuclear_Foci <- cbind(df_nuclear_Foci, df_nuclear_Size)


df_nuclear_Foci2 <- read.csv(main_foci_file2)
df_nuclear_Size2 <- read.csv(nuc_file2)
df_nuclear_Foci2 <- cbind(df_nuclear_Foci2, df_nuclear_Size2)

dim(df_nuclear_Foci)
dim(df_nuclear_Size)


#Combine DataFrames:
df_nuclear_Foci <- rbind(df_nuclear_Foci, df_nuclear_Foci2)
df_nuclear_Foci <- df_nuclear_Foci[ , c("ImageNumber", "FileName_DAPI", "Children_DsRed_foci_Count", "Children_GFP_foci_Count", "AreaShape_Area","AreaShape_Compactness", "AreaShape_FormFactor")]
#Renames the FileName_Hela_cells --> "Names"
colnames(df_nuclear_Foci)[2] <- "Names"
colnames(df_nuclear_Foci)[3] <- "DsRed"
colnames(df_nuclear_Foci)[4] <- "GFP"
colnames(df_nuclear_Foci)[5] <- "NucleusArea"
colnames(df_nuclear_Foci)[6] <- "Compactness"
colnames(df_nuclear_Foci)[7] <- "FormFactor"


sub_df <- df_nuclear_Foci

replace_names_2 <- function(x) {
  x <- gsub(".*(Celastrol_3uM_3h_MLF2GFP_K48).*", "Celastrol_3 µM_MLF2-GFP_K48-Ubiquitin", x)
  x <- gsub(".*(Celastrol_3uM_3h_endogenousMLF2).*", "Celastrol_3 µM_Endogenous MLF2", x)
  x <- gsub(".*(celastrol_3uM_3h_HSPA1A_488_DNAJB2).*", "Celastrol_3 µM_HSPA1A_DNAJB2", x)

  x <- gsub(".*(PZ3_3uM_3h_MLF2GFP_K48).*", "Pyrithione Zinc_3 µM_MLF2-GFP_K48-Ubiquitin", x)
  x <- gsub(".*(PZ_3uM_3h_HSPA1A_488_DNAJB2).*", "Pyrithione Zinc_3 µM_HSPA1A_DNAJB2", x)
  x <- gsub(".*(PZ_3uM_3h_endogenousMLF2_488).*", "Pyrithione Zinc_3 µM_Endogenous MLF2", x)

  x <- gsub(".*(Disulfirum_50uM_3h_HSPA1A_488_DNAJB2).*", "Disulfiram_50 µM_HSPA1A_DNAJB2", x)
  x <- gsub(".*(Disulfirum_50uM_3h_MLF2GFP_K48).*", "Disulfiram_50 µM_MLF2-GFP_K48-Ubiquitin", x)
  x <- gsub(".*(Disulfirum_50uM_3h_endogenousMLF2_488).*", "Disulfiram_50 µM_Endogenous MLF2", x)

  x <- gsub(".*(Plumbagin_15uM_3h_HSPA1A_488_DNAJB2).*", "Plumbagin_15 µM_HSPA1A_DNAJB2", x)
  x <- gsub(".*(Plumbagin_15uM_3h_MLF2GFP_K48).*", "Plumbagin_15 µM_MLF2-GFP_K48-Ubiquitin", x)
  x <- gsub(".*(Plumbagin_15uM_3h_endogenousMLF2_488).*", "Plumbagin_15 µM_Endogenous MLF2", x)

  x <- gsub(".*(DMSO_3h_MLF2GFP_K48).*", "DMSO_DMSO_MLF2-GFP_K48-Ubiquitin", x)
  x <- gsub(".*(DMSO_3h_endogenousMLF2_488).*", "DMSO_DMSO_Endogenous MLF2", x)
  x <- gsub(".*(DMSO_3h_HSPA1A_488_DNAJB2).*", "DMSO_DMSO_HSPA1A_DNAJB2", x)

}

sub_df <- df_nuclear_Foci
sub_df$Names <- sapply(sub_df$Names, replace_names_2)

sub_df2 <- sub_df[sub_df$Names != "NA", ]



sub_df2 <- sub_df %>%
  separate(Names, into = c("Drug", "Treatment", "GFP Marker", "DsRed Marker"), sep = "_")

sub_GFP <- sub_df2[, c("ImageNumber", "Treatment", "GFP Marker", "GFP", "NucleusArea", "Compactness", "FormFactor", "Drug" )]
sub_GFP$Foci_Type <- "GFP"
colnames(sub_GFP)[3] <- "Marker"
colnames(sub_GFP)[4] <- "Foci"

sub_dsRed <- sub_df2[, c("ImageNumber", "Treatment", "DsRed Marker", "DsRed", "NucleusArea", "Compactness", "FormFactor", "Drug")]
sub_dsRed$Foci_Type <- "dsRed"
colnames(sub_dsRed)[3] <- "Marker"
colnames(sub_dsRed)[4] <- "Foci"

combined_Df <- rbind(sub_GFP, sub_dsRed)


combined_Df$Drug <- factor(combined_Df$Drug, levels =  c("DMSO", "Pyrithione Zinc", "Celastrol", "Plumbagin", "Disulfiram" ))

library(dplyr)
combined_Df$Drug <- combined_Df$Drug %>%
  recode_factor(
    "DMSO" = "DMSO",
    "Pyrithione Zinc" = "Pyrithione Zinc, 3 µM",
    "Celastrol" = "Celastrol, 3 µM",
    "Plumbagin" = "Plumbagin, 15 µM",
    "Disulfiram" = "Disulfiram, 50 µM"
  )

combined_Df$Marker <- factor(combined_Df$Marker, levels = c("MLF2-GFP", "K48-Ubiquitin"))
combined_Df <- na.omit(combined_Df[ combined_Df$Treatment != "NA", ])

combined_Df <- combined_Df %>%
  mutate(Drug = as.factor(Drug))


#Confidential Naming:
combined_Df$Drug <- combined_Df$Drug %>%
  recode_factor(
    "DMSO" = "DMSO",
    "Pyrithione Zinc, 3 µM" = "DP-2303",
    "Celastrol, 3 µM" ="DP-665",
   "Plumbagin, 15 µM" =  "DP-1117",
     "Disulfiram, 50 µM" = "DP-1699"
  )  

combined_Df$Drug <- factor(combined_Df$Drug, levels =  c("DMSO", "DP-2303", "DP-665", "DP-1117", "DP-1699" ))


stat.test <- combined_Df %>%
  group_by(Marker) %>%
  wilcox_test(Foci ~ Drug, ref.group = "DMSO") %>%
  adjust_pvalue(method = "bonferroni")

stat.test <- stat.test %>%
  add_xy_position(x = c("Marker"))

y_positions <- seq(from = max(combined_Df$Foci) / 1.2, to = max(combined_Df$Foci) , length.out = 4)
y_positions <- c(y_positions, y_positions)
stat.test <- stat.test %>%
  mutate(y.position = rep(y_positions)) 


#NEED THIS OR WILCOX DOES NOT WORK 
stat.test$Drug <- stat.test$group2



library(viridis)           
          scale_fill_viridis(discrete = TRUE, option = "B", direction =  -1,  begin = 0, end = 0.75) 

  
  jittered_BoxPlot <- function(variable_of_Interests, plot_Title, y_Axis_Label, variable_2){
    inital_Plot <- ggplot(combined_Df, aes(x = Marker,  y = !!rlang::sym(variable_of_Interests), fill = Drug)) + 
   
      geom_jitter(position = position_jitterdodge(jitter.width = 0.1, dodge.width = 0.75), size = 2, show.legend = TRUE) +
      
      geom_boxplot(alpha = 0.5, width = 0.75) +

      theme_classic() +
       stat_pvalue_manual(stat.test, label = "p.adj.signif", tip.length = 0.01, size = 7.5) +
      labs(y = y_Axis_Label, title = plot_Title, fill = "Conditions:") + 
      theme(axis.title.y = element_text(size = 18), axis.title.x = element_blank()) +
        theme(axis.text.x = element_text(size = 30, angle = 55, hjust = 1, face = "bold")) +
       theme(legend.text = element_text(size = 24, face = "bold"),
          legend.title = element_text(size = 24, face = "bold")) +
    
   
      theme(legend.position = c(0.875, 0.7), legend.background = element_blank()) + # Adjusted for vertical centering
      ylim(0, max(combined_Df[[variable_of_Interests]])) +
      scale_x_discrete(expand = expansion(add = c(-1.5, 0))) +
      
    theme(axis.text.y = element_text(size = 22, face = "bold"), # Increase y-axis label size
    axis.title.y = element_text(size = 34, face = "bold"),
      plot.title = element_blank()) +
      scale_fill_viridis(discrete = TRUE, option = "B", direction =  -1,  begin = 0, end = 0.75) 


      
  }
  
        
  
  p1 <- jittered_BoxPlot("Foci", "Pharmacological Reduction of NE Condensate Markers:", "NE Condensate Foci / Cell", "sd_MLF2")
  p1
  
tiff("C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/grants_drafts/poch_2024/MLF2K48Boxplots.tif", units="in", width=10, height=10, res=300)
ggarrange(p1, heights = c(10, 10))
dev.off()

ggsave(ggsave("C:/Users/Schlieker Lab/Desktop/New_Data_AFTER_23_11_17/Dylan_new/grants_drafts/poch_2024/MLF2K48Boxplots.PDF"), p1, dpi = 300, width = 10, height = 10)

```

